modules/auxiliary/gather/memcached_extractor.rb

```
@@ -0,0 +1,94 @@
+        finds the keys and values stored in those slabs.
+      ),
+      'Author'       => [ 'Paul Deardorff <paul_deardorff[at]rapid7.com>' ],
+      'License'      => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(11211)
+      ], self.class)
+  end
+
+  # Returns array of keys for all slabs
+  def enumerate_keys
+    keys = []
+    enumerate_slab_ids.each do |sid|
+      sock.send("stats cachedump #{sid} 100\r\n", 0)
+      data = sock.recv(4096)
+      matches = /^ITEM (?<key>.*) \[/.match(data)
+      keys << matches[:key] if matches
+    end
+    keys
+  end
+
+  # Returns array of slab ids as strings
+  def enumerate_slab_ids
+    sock.send("stats slabs\r\n", 0)
+    slab_ids = []
+    loop do
+      data = sock.recv(4096)
+      break if !data || data.length == 0
+      matches = data.scan(/^STAT (?<slab_id>(\d)*):/)
+      slab_ids << matches
+      break if data =~ /^END/
+    end
+    slab_ids.flatten!
+    slab_ids.uniq! || []
+  end
+
+  def data_for_keys(keys = [])
+    all_data = {}
+    keys.each do |key|
+      sock.send("get #{key}\r\n", 0)
+      data = []
+      loop do
+        data_part = sock.recv(4096)
+        break if !data_part || data_part.length == 0
+        data << data_part
+        break if data_part =~ /^END/
+      end
+      all_data[key] = data
+    end
+    all_data
+  end
+
+  def determine_version
+    sock.send("stats\r\n", 0)
+    stats = sock.recv(4096)
+    matches = /^STAT (?<version>version (\.|\d)*)/.match(stats)
+    matches[:version] || 'unkown version'
+  end
+
+  def run
+    print_status("#{rhost}:#{rport} - Connecting to memcached server...")
+    begin
+      connect
+      print_good("Connected to memcached #{determine_version}")
+      keys = enumerate_keys
+      print_good("Found #{keys.size} keys")
+      data = data_for_keys(keys)
+      rhost = 'localhost.memcached' if %w(localhost 127.0.0.1).include?(rhost)
```

Sorry, what I mean to convey was that I know why you are doing it, but this is a known limitation of the model and no modules are able to report on localhost data.  Excluding the fact that rhost should never be changed (it is set for you), even if you didn't modify rhost but instead used a ternary operator in your store_loot call, you'd effectively be working around something that isn't expected to work anyway.   If someone set RHOST to 127.0.0.1, any module that stores data for that RHOST will fail.

You could also:

1 Check ```rhost``` for localhost/127.0.0.1 and not store_loot if so
2 Allow ```rhost``` to be localhost/127.0.0.1 and let it blow up naturally
3 1 + printing the keys in a Rex::UI::Table or some such thing